### ISSUE TITLE
OpenSSL + MSN: crashes at "stack.c (326): sk_pop_free",  "netlibhttp.cpp (889): Netlib_HttpTransaction", "msn_ssl.cpp (79): CMsnProto::getSslResult" and "msn_auth.cpp (782): CMsnProto::RefreshOAuth"

### ISSUE THREAD

**Issue by Corak667** [miranda-ng/miranda-ng#832]
Crashlog:
[Miranda_OpenSSL-MSN_sk_pop_free_20170528.zip](https://github.com/miranda-ng/miranda-ng/files/1034133/Miranda_OpenSSL-MSN_sk_pop_free_20170528.zip)

First crash:
```
Likely cause of the crash plugin: OpenSSL SSL API module

Exception: Access Violation at address 121E979C. Reading from address BABA0054.

Stack Trace:
---------------------------------------------------------------
121E979C (LIBEAY32 12180000): w:\test\openssl\crypto\stack\stack.c (325): sk_pop_free
12222045 (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (148): x509_verify_param_zero
12222130 (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (203): X509_VERIFY_PARAM_free
126644FF (SSLEAY32 12640000): w:\test\openssl\ssl\ssl_lib.c (583): SSL_free
1630100E (OpenSSL 16300000): (filename not available) (0): (function-name not available)
12444B14 (mir_app 12420000): c:\sources\miranda_ng_compilation\src\mir_app\src\netlibhttp.cpp (889): Netlib_HttpTransaction
15FE041C (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_ssl.cpp (79): CMsnProto::getSslResult
15FE1BC3 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_soapstore.cpp (244): CMsnProto::MSN_StoreGetProfile
15FD9028 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_contact.cpp (303): CMsnProto::MSN_RefreshContactList
15FC7169 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_threads.cpp (82): CMsnProto::msn_loginThread
125C7F67 (mir_core 125C0000): c:\sources\miranda_ng_compilation\src\mir_core\src\threads.cpp (159): forkthreadex_r
7C80B729 (kernel32 7C800000): (filename not available) (0): GetModuleFileNameA
```

second crash (after 5 hours (didnt closed the miranda):

```
Likely cause of the crash plugin: OpenSSL SSL API module

Exception: Access Violation at address 121E97A8. Reading from address 313A2274.

Stack Trace:
---------------------------------------------------------------
121E97A8 (LIBEAY32 12180000): w:\test\openssl\crypto\stack\stack.c (326): sk_pop_free
1222205F (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (153): x509_verify_param_zero
12222130 (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (203): X509_VERIFY_PARAM_free
126644FF (SSLEAY32 12640000): w:\test\openssl\ssl\ssl_lib.c (583): SSL_free
1630100E (OpenSSL 16300000): (filename not available) (0): (function-name not available)
12444B14 (mir_app 12420000): c:\sources\miranda_ng_compilation\src\mir_app\src\netlibhttp.cpp (889): Netlib_HttpTransaction
15FDE485 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (782): CMsnProto::RefreshOAuth
15FDEB49 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (236): OAuthToken::Refresh
15FDE191 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (1041): CMsnProto::MSN_RefreshOAuthTokens
125C7ECE (mir_core 125C0000): c:\sources\miranda_ng_compilation\src\mir_core\src\threads.cpp (327): Thread_Push

```

**Comment by Corak667** [comment]
+ Crashed again, but from address 00000000:
```

Likely cause of the crash plugin: OpenSSL SSL API module

Exception: Access Violation at address 121E97A8. Reading from address 00000000.

Stack Trace:
---------------------------------------------------------------
121E97A8 (LIBEAY32 12180000): w:\test\openssl\crypto\stack\stack.c (326): sk_pop_free
1222205F (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (153): x509_verify_param_zero
12222130 (LIBEAY32 12180000): w:\test\openssl\crypto\x509\x509_vpm.c (203): X509_VERIFY_PARAM_free
126644FF (SSLEAY32 12640000): w:\test\openssl\ssl\ssl_lib.c (583): SSL_free
1630100E (OpenSSL 16300000): (filename not available) (0): (function-name not available)
12444B14 (mir_app 12420000): c:\sources\miranda_ng_compilation\src\mir_app\src\netlibhttp.cpp (889): Netlib_HttpTransaction
15FDE485 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (782): CMsnProto::RefreshOAuth
15FDEB49 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (236): OAuthToken::Refresh
15FDE191 (MSN 15FC0000): c:\sources\miranda_ng_compilation\protocols\msn\src\msn_auth.cpp (1041): CMsnProto::MSN_RefreshOAuthTokens
125C7ECE (mir_core 125C0000): c:\sources\miranda_ng_compilation\src\mir_core\src\threads.cpp (327): Thread_Push

```

http://coraksoft1.narod.ru/temp/miranda/Miranda_OpenSSL-MSN_sk_pop_free_RefreshOAuth_20170529.zip

**Comment by sagamusix** [comment]
Got this as well:
```

Exception: Access Violation at address 0000000051FDA0E5. Reading from address 0000000000000356.

Stack Trace:
---------------------------------------------------------------
0000000051FDA0E5 (LIBEAY32 0000000051F30000): (filename not available) (0): sk_pop_free
0000000052021FA4 (LIBEAY32 0000000051F30000): (filename not available) (0): X509_LOOKUP_hash_dir
00000000520220CB (LIBEAY32 0000000051F30000): (filename not available) (0): X509_VERIFY_PARAM_free
0000000051EF9F55 (SSLEAY32 0000000051ED0000): (filename not available) (0): SSL_free
0000000016A41E4C (OpenSSL 0000000016A40000): (filename not available) (0): (function-name not available)
0000000012652BAD (mir_app 0000000012620000): (filename not available) (0): Netlib_RegisterUser
000000001264CF11 (mir_app 0000000012620000): (filename not available) (0): Netlib_OpenConnection
0000000012650DE2 (mir_app 0000000012620000): (filename not available) (0): Netlib_SendHttpRequest
0000000012650B56 (mir_app 0000000012620000): (filename not available) (0): Netlib_HttpTransaction
00000000166A704C (MSN 0000000016680000): (filename not available) (0): (function-name not available)
00000000166A787F (MSN 0000000016680000): (filename not available) (0): (function-name not available)
00000000166A6C8B (MSN 0000000016680000): (filename not available) (0): (function-name not available)
0000000016688933 (MSN 0000000016680000): (filename not available) (0): (function-name not available)
000000001280A71C (mir_core 0000000012800000): (filename not available) (0): Thread_Wait
0000000076C859CD (kernel32 0000000076C70000): (filename not available) (0): BaseThreadInitThunk
0000000076EBA561 (ntdll 0000000076E90000): (filename not available) (0): RtlUserThreadStart
```

**Comment by dartraiden** [comment]
Closed in favor of #930
